plugins/newsql/newsql.c

```diff
@@ -2210,6 +2217,24 @@ static CDB2QUERY *read_newsql_query(struct dbenv *dbenv,
         goto retry_read;
     }
 
+    if (gbl_uses_externalauth && !identity_dbinfo_sent &&
+        query->sqlquery && !query->sqlquery->identity) {
+        int client_supports_identity = 0;
+        for (int ii = 0; ii < query->sqlquery->n_features; ++ii) {
```

Comment:
If the client doesn't support identity, does this loop always run?

Reply:
we don't  'goto retry_read' if client doesn't support the feature

Reply:
but we still check all the features, if server expects identity (uses_externalauth), but client hasn't sent it..